src/devices/nice_flor_s.c

```diff
@@ -37,29 +41,45 @@ static int nice_flor_s_decode(r_device *decoder, bitbuffer_t *bitbuffer)
     bitbuffer_invert(bitbuffer);
     uint8_t *b = bitbuffer->bb[0];
 
-    uint8_t button_id = b[0] >> 4;
-    if (button_id < 1 || button_id > 4) {
-        return DECODE_ABORT_EARLY;
+    uint8_t encbuff[7];
+    encbuff[0] = (b[0] >> 4) & 0x0f;
+    for (uint8_t i = 0; i < 6; i++) {
+        encbuff[i+1] = ((b[i] << 4) & 0xf0) | ((b[i+1] >> 4) & 0x0f);
     }
-    int count = 1 + (((b[0] ^ ~button_id) - 1) & 0xf);
-    uint32_t serial = ((b[1] & 0xf0) << 20) | ((b[3] & 0xf) << 20) |
-       (b[4] << 12) | (b[5] << 4) | (b[6] >> 4);
-    uint16_t code = (b[1] << 12) | (b[2] << 4) | (b[3] >> 4);
+
+    uint16_t enccode = ((encbuff[2] << 8) & 0xff00) | (encbuff[3] & 0x00ff);
+    uint16_t deccode = nice_flor_s_table_decode[enccode];
+    uint8_t ki = nice_flor_s_table_ki[deccode & 0xff] ^ (enccode & 0xff);
+
+    uint8_t       snbuff[4];
+    snbuff[3] = (encbuff[1] ^ ki) & 0x0f;
+    snbuff[2] = encbuff[4] ^ ki;
+    snbuff[1] = encbuff[5] ^ ki;
+    snbuff[0] = encbuff[6] ^ ki;
+
+    uint32_t serial = (uint32_t)((snbuff[3] << 24) & 0xff000000) | ((snbuff[2] << 16) & 0xff0000) | ((snbuff[1] << 8) & 0xff00) | (snbuff[0] & 0xff);
+    uint16_t code = deccode;
+    uint8_t button_id = encbuff[0] & 0x0f;
+    uint8_t repeat = ((encbuff[1] >> 4) & 0x0f) ^ (encbuff[0] & 0x0f) ^ 0x0f;
+
 
     /* clang-format off */
     data_t *data = data_make(
-            "model",  "",              DATA_STRING, "Nice-FlorS",
+            "model",  "",              DATA_STRING, "Nice Flor-S",
             "button", "Button ID",     DATA_INT,     button_id,
-            "serial", "Serial (enc.)", DATA_FORMAT, "%07x",        DATA_INT, serial,
-            "code",   "Code (enc.)",   DATA_FORMAT, "%04x",        DATA_INT, code,
-            "count",  "",              DATA_INT,     count,
+            "serial", "Serial",        DATA_FORMAT, "0x%07x", DATA_INT, serial,
```

Comment:
Format here is generally only for nicer screen output and should not include a `0x` prefix.